### PR TITLE
DYN-5296 setting dynamic height for notification center popup

### PR DESCRIPTION
**Pull request by Enzo707**
### Purpose

https://github.com/DynamoDS/Dynamo/assets/111511512/9b0c5cae-63b3-489e-a081-6ddabc494dec

This PR is related to [DYN-5296](https://jira.autodesk.com/browse/DYN-5296)

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

This PR implements a dynamic height resizing for notifications popup base on Notification Center height.

### Reviewers

@RobertGlobant20 

### FYIs

@RobertGlobant20 
@reddyashish

### CODE REVIEW

**Comment by QilongTang** [comment]
Rebase your branch? @Enzo707

**Comment by reddyashish** [comment]
PR on notification center side: https://github.com/DynamoDS/NotificationCenter/pull/36

**Comment by QilongTang** [comment]
hi @Enzo707 @reddyashish The PR https://github.com/DynamoDS/hig/pull/3 has been merged, right now we only need to publish the npm packages for Dynamo repo to consume

**Comment by QilongTang** [comment]
hi @Enzo707 the reported test failure is a sporadic one which is going to be handled by @avidit , please address the PR comments and we can wrap up this PR

**Comment by QilongTang** [comment]
hi @Enzo707 While building your PR locally I did not get the same results, can you double check? Also cleaned up one function that is not used
image

File: src/Notifications/NotificationCenterController.cs [hunk 1]
```diff
@@ -24,16 +25,23 @@ namespace Dynamo.Notifications
...
             this.onMarkAllAsRead = onMarkAllAsRead;
+            this.onNotificationPopupUpdated = onNotificationPopupUpdated;
         }
 
         public void SetNoficationsAsRead(object[] ids)
         {
             onMarkAllAsRead(ids);
         }
+
+        public void UpdateNotificationWindowSize(int height)
```
[the remark this answers is not in the document]
Comment (QilongTang) [reply src/Notifications/NotificationCenterController.cs:41]: For function comments, please use /// while // is mostly used for inline comments for block of code

[the remark this answers is not in the document]
Comment (Enzo707) [reply src/Notifications/NotificationCenterController.cs:41]: @QilongTang got it! Thanks

**Comment by QilongTang** [comment]
EngOps build error:
"E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj" (Rebuild target) (96) ->
(AfterBuildOps target) -> 
  E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: The command "C:\Users\svc_p_dynamobuild\.nuget\packages\ilmerge\3.0.41\build\..\tools\net452\ILMerge.exe bin\AnyCPU\Release\Md2Html.exe ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		/allowDup ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		/lib:bin\AnyCPU\Release\ ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		/out:E:\Builds\Dynamo_master\Dynamo\src\..\bin\AnyCPU\Release\Md2Html\Md2Html.exe ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		AngleSharp.Css.dll ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		AngleSharp.dll ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		HtmlSanitizer.dll ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		Markdig.dll ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		NDesk.Options.dll ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		System.Buffers.dll ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		System.Memory.dll ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		System.Numerics.Vectors.dll ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		System.Runtime.CompilerServices.Unsafe.dll ^
E:\Builds\Dynamo_master\Dynamo\src\Tools\Md2Html\Md2Html.csproj(41,5): error MSB3073: 		System.Text.Encoding.CodePages.dll " exited with code 1.

    5766 Warning(s)
    1 Error(s)

**Pull request merged**